spanner/sessionclient.go

```diff
@@ -130,10 +132,32 @@ func (sc *sessionClient) createSession(ctx context.Context) (*session, error) {
 		return nil, err
 	}
 	ctx = contextWithOutgoingMetadata(ctx, sc.md)
+	var md metadata.MD
 	sid, err := client.CreateSession(ctx, &sppb.CreateSessionRequest{
 		Database: sc.database,
 		Session:  &sppb.Session{Labels: sc.sessionLabels},
-	})
+	}, gax.WithGRPCOptions(grpc.Header(&md)))
+
```

Comment:
Check the error from `CreateSession` and return that if it is non-nil before trying to extract the metadata.